javascript/javascript3/week3/Shopping Cart/shoppingCart.js

```diff
@@ -0,0 +1,135 @@
+const reducer = (acc, currentVal) => acc + currentVal;
+const apiKey = "8df58438c8513b601e88d88777ea74e8";
+
+const arrayOfProducts = [];
+
+const listItem1 = document.getElementById("list-item-1");
+const listItem2 = document.getElementById("list-item-2");
+
+const userNamePlace = document.getElementById("user-name-spot");
+const pricePlace = document.getElementById("total-price");
+
+class Product {
+  constructor(name, price) {
+    this.name = name;
+    this.price = price;
+  }
+
+  convertToCurrency(currency) {
+    //base currency is euro
+    const currencyConventerAPI = `http://data.fixer.io/api/latest?access_key=${apiKey}`;
+    fetch(currencyConventerAPI)
+      .then(res => res.json())
+      .then(data =>
+        console.log(
+          `${this.name} is ${Math.round(
+            this.price * data.rates[currency]
+          )} ${currency}`
+        )
+      );
+  }
+}
+
+const gum = new Product("gum", 2);
+gum.convertToCurrency("DKK");
+gum.convertToCurrency("USD");
+gum.convertToCurrency("ISK");
+
+class ShoppingCart {
+  constructor(products) {
+    this.products = products;
+  }
+
+  addProduct(product) {
+    this.products.push(product);
+  }
+
+  removeProduct(product) {
+    this.products.forEach(item => {
+      if (this.products.indexOf(product) !== -1) {
+        const filtered = this.products.splice(
+          this.products.indexOf(product),
+          1
+        );
+        return filtered;
+      }
+    });
+  }
+
+  searchProduct(productName) {
+    const product = [];
+    this.products.forEach(item => {
+      if (item.name.includes(productName)) {
+        return product.push(item);
+      }
+    });
+    console.log(product);
+    return product;
+  }
+
+  getTotal() {
+    const productPrices = [];
+    this.products.forEach(item => productPrices.push(item.price));
+    const sum = productPrices.reduce(reducer);
+    pricePlace.innerText = `Total: ${sum}`;
+    sum;
```

Comment:
what does this do?

Reply:
I deleted the unnecessary Sum but do you think I should put dom manipulation into renderproducts?

Reply:
Yes DOM manipulation should go in the "render..." functions